src/Symfony/Component/Validator/Constraints/UniqueValidator.php

```diff
@@ -69,18 +80,37 @@ private function getNormalizer(Unique $unique): callable
         return $unique->normalizer;
     }
 
-    private function reduceElementKeys(array $fields, array $element): array
+    private function reduceElementKeys(array $fields, array|object $element): array
     {
         $output = [];
         foreach ($fields as $field) {
             if (!\is_string($field)) {
                 throw new UnexpectedTypeException($field, 'string');
             }
-            if (isset($element[$field])) {
-                $output[$field] = $element[$field];
+
+            // For no BC, because PropertyAccessor require brackets for array keys
+            // Previous implementation, only check in array
+            if (\is_array($element) && !str_contains($field, '[')) {
```

Comment:
I don't think validating deeper than first level should be allowed.
Otherwise we may end up with validating uniqueness of other object.